From 5a2cecce802911c2b2f3cf323141cc32ee5d89ed Mon Sep 17 00:00:00 2001
From: Stanislav Kholmanskikh <stanislav.kholmanskikh@bell-sw.com>
Date: Mon, 11 Dec 2023 13:07:13 +0300
Subject: [PATCH] Set default rc_cgroup_mode back to hybrid

When it's "unified", docker top fails. See:

https://github.com/opencontainers/runc/issues/4097

It looks like an issue in runc, but fixing it
may take some time. Meanwhile we should make docker
work with the default openrc configuration.
---
 etc/rc.conf       | 2 +-
 init.d/cgroups.in | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/etc/rc.conf b/etc/rc.conf
index d359eb9..0029d51 100644
--- a/etc/rc.conf
+++ b/etc/rc.conf
@@ -199,7 +199,7 @@ rc_tty_number=12
 # cgroups version 1 on /sys/fs/cgroup.
 # "legacy" mounts cgroups version 1 on /sys/fs/cgroup
 # "unified" mounts cgroups version 2 on /sys/fs/cgroup
-#rc_cgroup_mode="unified"
+#rc_cgroup_mode="hybrid"
 
 # This is a list of controllers which should be enabled for cgroups version 2
 # when hybrid mode is being used.
diff --git a/init.d/cgroups.in b/init.d/cgroups.in
index 218cce7..be49a4c 100644
--- a/init.d/cgroups.in
+++ b/init.d/cgroups.in
@@ -87,7 +87,7 @@ cgroup2_controllers()
 	[ ! -e "${cgroup_path}/cgroup.subtree_control" ]&& return 0
 	read -r active < "${cgroup_path}/cgroup.controllers"
 	for x in ${active}; do
-	case "${rc_cgroup_mode:-unified}" in
+	case "${rc_cgroup_mode:-hybrid}" in
 		unified)
 			echo "+${x}"  > "${cgroup_path}/cgroup.subtree_control"
 			;;
@@ -128,7 +128,7 @@ cgroups_unified()
 
 mount_cgroups()
 {
-	case "${rc_cgroup_mode:-unified}" in
+	case "${rc_cgroup_mode:-hybrid}" in
 	hybrid) cgroups_hybrid ;;
 	legacy) cgroups_legacy ;;
 	unified) cgroups_unified ;;
-- 
2.43.0

